guildwars2/characters.py

```diff
@@ -256,6 +257,229 @@ def handle_duplicates(upgrades):
+                        attr_dict["Healing"] += int(modifier)
+                        attr_dict["ConditionDamage"] += int(modifier)
+                    elif pattern_single.match(bonus):
+                        # Regex deciding the attribute name + modifier
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attribute_name = re.sub('^.* ', '', bonus)
+                        if attribute_name in attr_dict:
+                            attr_dict[attribute_name] += int(modifier)
+                    elif pattern_percentage.match(bonus):
+                        modifier = re.sub(' .*$', '', bonus)
+                        modifier = re.sub('\+', '', modifier)
+                        attribute_name = re.sub(' Duration', 'Duration', bonus)
+                        attribute_name = re.sub('^.* ', '', attribute_name)
+                        if attribute_name in attr_dict:
+                            attr_dict[attribute_name] += int(modifier)
+                    # Amount of runes equipped
+                    count += 1
+
+        # Calculate base value depending on char level
+        basevalue = self.calcBaselvl(level, 0, lvl_dict)
+        attr_dict["Power"] += basevalue
+        attr_dict["Vitality"] += basevalue
+        attr_dict["Toughness"] += basevalue
+        attr_dict["Precision"] += basevalue
+        # Mapping for old attribute names
+        attr_dict["Concentration"] += attr_dict["BoonDuration"]
+        attr_dict["Ferocity"] += attr_dict["CritDamage"]
+        attr_dict["Expertise"] += attr_dict["ConditionDuration"]
+        # Calculate derivative attributes
+        # Reset to default after mapped to new attribute name
+        attr_dict["CritDamage"] = 150 + round(attr_dict["Ferocity"] / 15, 2)
+        attr_dict["BoonDuration"] = round(attr_dict["Concentration"] / 15, 2)
+        attr_dict["ConditionDuration"] = round(attr_dict["Expertise"] / 15, 2)
+        # Base value of 1000 on lvl 80 doesn't get calculated, if below lvl 80 dont subtract it
+        if attr_dict["Precision"] < 1000:
+            base_prec = 0
+        else:
+            base_prec = 1000
+        attr_dict["Critical Chance"] = 4 + round(
+            (attr_dict["Precision"] - base_prec) / 21, 2)
+        attr_dict["defense"] += attr_dict["Toughness"]
+
+        # Calculate base health
+        attr_dict["Health"] = self.calcBaseHealth(level, 0,
+                                                  profession_group[profession])
+        attr_dict["Health"] += attr_dict["Vitality"] * 10
+
+        ordered_list = ('Power', 'Toughness', 'defense', 'Vitality', 'Health',
+                        'Precision', 'Critical Chance', 'Ferocity',
+                        'CritDamage', 'ConditionDamage', 'Healing',
+                        'Expertise', 'ConditionDuration', 'Concentration',
+                        'BoonDuration', 'AgonyResistance')
```

Comment:
Should probably use some re.sub to insert a space before capital letters, and use str.title() to capitalize properly for better display

Reply:
Added .capitalize()
Added regex for substitution